integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-method.component.ts

```diff
@@ -0,0 +1,70 @@
+import { DP_CARD_REGISTRATION_STATUS } from '../../../utils/dp-constants';
+import { ActivatedRoute } from '@angular/router';
+import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
+import {
+  CheckoutStepService,
+  PaymentMethodComponent as CorePaymentMethodComponent,
+} from '@spartacus/checkout/components';
+import {
+  UserPaymentService,
+  GlobalMessageService,
+  TranslationService,
+  ActiveCartService,
+  PaymentDetails,
+} from '@spartacus/core';
+import {
+  CheckoutService,
+  CheckoutDeliveryService,
+  CheckoutPaymentService,
+} from '@spartacus/checkout/core';
+
+@Component({
+  selector: 'cx-payment-method',
+  templateUrl: './dp-payment-method.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DpPaymentMethodComponent
+  extends CorePaymentMethodComponent
+  implements OnInit {
+  showCallbackScreen = false;
+
+  isDpCallback(): boolean {
+    const queryParams = this.activatedRoute.snapshot.queryParams;
+    return DP_CARD_REGISTRATION_STATUS in queryParams;
+  }
+
+  hideCallbackScreen(): void {
+    this.showCallbackScreen = false;
+  }
+
+  paymentDetailsAdded(paymentDetails: PaymentDetails) {
```

Comment:
Can we have some unit tests for this extra code?

Reply:
done